Startup.cs

```diff
@@ -32,7 +37,18 @@ public void ConfigureServices(IServiceCollection services)
                 config.UseSqlServer(_config.GetConnectionString("FitnesssTrackerConn"));
             });
 
+            services.AddAutoMapper(Assembly.GetExecutingAssembly());
+
+            services.AddScoped<ILogRepository, LogRepository>();
+            services.AddScoped<IExerciseRepository, ExerciseRepository>();
+            services.AddScoped<IUserRepository, UserRepository>();
+            services.AddScoped<ILogService, LogService>();
+
             services.AddControllers();
+            services.AddMvc().AddNewtonsoftJson( opt =>
+            {
+                opt.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
```

Comment:
maybe look into [JSONIgnore] attributes instead of a global setting to avoid loops

Reply:
removed bi directional ref by mapping to a DTO using automapper